docs/building-blocks/extending-templates/collector.md

```diff
@@ -0,0 +1,309 @@
+
+To work around this, you can implement a property transform and collector that iterates your object array and transforms it into the JSON schema expected by the resource.
+
+> [!IMPORTANT]
+> This approach requires that you have a deep understanding of resource manager templates and functions.
+
+Let's take a look at how we can implement a property collector and transformer with an example that deploys a [network security group (NSG)][nsg]. The diagram below shows the relationship between our templates and our resources within those templates:
+
+![property collector and transformer architecture](../_images/collector-transformer.png)
+
+Our **calling template** imcludes two resources:
+1) a template link that invokes our **collector template**, and
+2) a network security group (NSG).
+
+Our **collector template** includes two resources:
+1) an [**anchor** resource][anchor-resource], and
+2) a template link that invokes the transform template in a copy loop.
+
+Our **transform template** includes a single resource: an empty template with a variable that transforms our `source` JSON to the JSON schema expected by our NSG resource in the **main template**.
+
+## Parameter object
+
+We'll be using our `"securityRules"` parameter object from [objects as parameters][objects-as-parameters]. Our **transform template** will transform each object in the `"securityRules"` array into the JSON schema expected by the NSG resource in our **calling template**.
+
+```json
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters":{ 
+      "networkSecurityGroupsSettings": {
+      "value": {
+          "securityRules": [
+            {
+              "name": "RDPAllow",
+              "description": "allow RDP connections",
+              "direction": "Inbound",
+              "priority": 100,
+              "sourceAddressPrefix": "*",
+              "destinationAddressPrefix": "10.0.0.0/24",
+              "sourcePortRange": "*",
+              "destinationPortRange": "3389",
+              "access": "Allow",
+              "protocol": "Tcp"
+            },
+            {
+              "name": "HTTPAllow",
+              "description": "allow HTTP connections",
+              "direction": "Inbound",
+              "priority": 200,
+              "sourceAddressPrefix": "*",
+              "destinationAddressPrefix": "10.0.1.0/24",
+              "sourcePortRange": "*",
+              "destinationPortRange": "80",
+              "access": "Allow",
+              "protocol": "Tcp"
+            }
+          ]
+        }
+      }
+    }
+  }
+```
```

Comment:
This is the parameter file, not the final JSON? It might be helpful to show the before and after?